modules/javafx.base/src/main/java/com/sun/javafx/binding/LazyObjectBinding.java

```diff
@@ -57,9 +57,9 @@ public void removeListener(ChangeListener<? super T> listener) {
 
     @Override
     public void addListener(InvalidationListener listener) {
-        super.addListener(listener);
+        updateSubscriptionBeforeAdd();
 
-        updateSubscriptionAfterAdd();
+        super.addListener(listener);
```

Comment:
I noticed that reverting this change causes all the `GivenAQuadMappedObservable` to fails, but reverting the add `ChangeListener` method change causes only change listener tests to fail. Is this correct? Shouldn't there be a tests that fails just the invalidation listener code?

Reply:
The lazy Bindings all use invalidation listeners internally, so there will always be some invalidation listeners in these cases.  There is the listener that the user adds (invalidation, change or the subscription variants that use one of these), and there are the 4 internal invalidation listeners needed to support the 4 mappings which are added and removed when observed only.